packages/backend-function/src/runtime/get_amplify_clients_configuration.ts

```diff
@@ -127,13 +139,14 @@ export const getAmplifyDataClientConfig = async <T>(
     modelIntrospectionSchema = JSON.parse(modelIntrospectionSchemaJson ?? '{}');
   } catch (caught) {
     if (caught instanceof NoSuchKey) {
-      throw new Error(
-        'Error retrieving the schema from S3. Please confirm that your project has a `defineData` included in the `defineBackend` definition.'
-      );
+      throw new AmplifyFault('S3ConfigFault', {
+        message:
+          'Error retrieving the schema from S3. Please confirm that your project has a `defineData` included in the `defineBackend` definition.',
+      });
     } else if (caught instanceof S3ServiceException) {
-      throw new Error(
-        `Error retrieving the schema from S3. You may need to grant this function authorization on the schema. ${caught.name}: ${caught.message}.`
-      );
+      throw new AmplifyFault('S3ConfigFault', {
+        message: `Error retrieving the schema from S3. You may need to grant this function authorization on the schema. ${caught.name}: ${caught.message}.`,
+      });
```

Comment:
Sorry I realized later that this code runs in lambda and not in the ampx binary. These should be fine with `Error`.

Reply:
I'll revert this back to using errors. Thanks for the update.